lib/cartopy/mpl/feature_artist.py

```diff
@@ -149,16 +173,22 @@ def draw(self, renderer, *args, **kwargs):
             extent = ax.get_extent(feature_crs)
         except ValueError:
             warnings.warn('Unable to determine extent. Defaulting to global.')
-        geoms = self._feature.intersecting_geometries(extent)
 
-        # Combine all the keyword args in priority order.
-        prepared_kwargs = style_merge(self._feature.kwargs,
-                                      self._kwargs,
-                                      kwargs)
+        if isinstance(self._feature, cfeature.ShapelyFeature):
+            # User passed a specific list of geometries.  If they also passed
+            # `array` or a list of facecolors then we should keep the colours
+            # consistent after pan/zoom.  Do this by creating a Path for every
+            # geometry regardless of whether they are currently in view.
+            geoms = self._feature.geometries()
```

Comment:
I thought about inserting empty paths for geometries that are not in view, which should be more efficient than transforming all of them.  However I'm not sure that the extra code complexity is justified: since the mapping from geometry to path is cached, it will only make a difference at the first draw.  Also if someone is passing a specific list of geometries then I would think they want to draw most/all of them at least once.

Reply:
It's a definite possibility that that would cause confusion!  But I suspect that the vast majority of users using Natural Earth or the other downloads would only pass one style for all geometries so will never notice the difference.